R/position-jitterdodge.R

```diff
@@ -35,17 +35,20 @@ PositionJitterdodge <- ggproto("PositionJitterdodge", Position,
   jitter.height = NULL,
   dodge.width = NULL,
 
-  required_aes = c("x", "y", "fill"),
+  required_aes = c("x", "y"),
 
   setup_params = function(self, data) {
-    width <- self$jitter.width %||% resolution(data$x, zero = FALSE) * 0.4
-    # Adjust the x transformation based on the number of 'fill' variables
-    nfill <- length(levels(data$fill))
-
+      width <- self$jitter.width %||% resolution(data$x, zero = FALSE) * 0.4
+      # Adjust the x transformation based on the number of 'dodge' variables
+      dodgecols <- intersect(c("fill", "colour", "linetype", "shape", "size", "alpha"), colnames(data))  
+      if (length(dodgecols) == 0) stop("position_jitterdodge requires at least aesthetic to dodge by", call. = FALSE)
```

Comment:
This should be spread over two lines, and please use ``"`position_jitterdodge()` requires...``